src/main/java/com/nike/cerberus/endpoints/secret/SecureDataEndpointV1.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    public final CompletableFuture<ResponseInfo<O>> execute(RequestInfo<I> request,
+                                                            Executor longRunningTaskExecutor,
+                                                            ChannelHandlerContext ctx) {
+
+        final Optional<SecurityContext> securityContext =
+                CmsRequestSecurityValidator.getSecurityContextForRequest(request);
+
+        // If the token was invalid throw Vault style permission denied error
+        if (! securityContext.isPresent() || ! (securityContext.get().getUserPrincipal() instanceof CerberusPrincipal)) {
+            log.error("Security context was null or principal was not instance of Cerberus Principal");
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        SecureDataRequestInfo requestInfo = parseInfoFromPath(request.getPath());
+
+        // if that path was invalid throw Vault style permission denied error
+        if (! doesRequestHaveRequiredParams(requestInfo.getCategory(), requestInfo.getSdbSlug())) {
+            log.error("Required path params missing, PATH: {}", request.getPath());
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        CerberusPrincipal principal = (CerberusPrincipal) securityContext.get().getUserPrincipal();
+
+        String sdbBasePath = String.format("%s/%s/", requestInfo.getCategory(), requestInfo.getSdbSlug());
+        Optional<String> sdbId = safeDepositBoxService.getSafeDepositBoxIdByPath(sdbBasePath);
+
+        // check that the principal has the proper permissions or throw Vault style permission denied error
+        if (! sdbId.isPresent() || ! permissionService.doesPrincipalHavePermission(principal, sdbId.get(),
+                        SecureDataAction.fromMethod(request.getMethod()))) {
+            log.info("SDB Id not found or permission was not granted for principal, principal: {}, sdb: {}," +
+                    " path: {}", principal.getName(), sdbId.orElseGet(() -> "missing"), request.getPath());
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
```

Comment:
Are we changing error codes?  I think vault gave 404 to avoid information disclosure if someone doesn't have permissions.

I like 404 for non-authed users especially

Reply:
You get a 403 if you try to access a path you do not have access to.
You get a 404 if you try to access a path you have access to but does not exist.

I manually tested both of those use cases against the existing API. Any deviations from this would be a breaking API change.

Reply:
Also if you have a bad token you also get a 403

Reply:
Adding link to this for the regression test ticket to verify these status codes for the edge cases.